document-generator-prosecution/src/main/java/uk/gov/companieshouse/document/generator/prosecution/UltimatumDocumentInfoBuilderProvider.java

```diff
@@ -0,0 +1,110 @@
+        templateRegistryAddress =
+                        environmentReader.getMandatoryString(ConfigKeys.TEMPLATE_REGISTRY_ADDRESS);
+    }
+
+    public UltimatumDocumentInfoBuilder builder() {
+        return new UltimatumDocumentInfoBuilder();
+    }
+
+    /**
+     * Create a {@link DocumentInfoResponse} for a Prosecution Ultimatum.
+     *
+     */
+    public class UltimatumDocumentInfoBuilder {
+        private ProsecutionCase prosecutionCase;
+        private String renderedDocFileName;
+
+        private UltimatumDocumentInfoBuilder() {}
+
+        /**
+         * Sets the prosecution case, a source of info for building the template values.
+         * @param prosecutionCase
+         * @return
+         */
+        public UltimatumDocumentInfoBuilder prosecutionCase(ProsecutionCase prosecutionCase) {
+            this.prosecutionCase = prosecutionCase;
+            return this;
+        }
+
+        /**
+         * Sets the name of file for the rendered doc.
+         * @param renderedDocFileName
+         * @return
+         */
+        public UltimatumDocumentInfoBuilder renderedDocFileName(String renderedDocFileName) {
+            this.renderedDocFileName = renderedDocFileName;
+            return this;
+        }
+
+        public DocumentInfoResponse build() throws DocumentInfoCreationException {
+            DocumentInfoResponse documentInfoResponse = new DocumentInfoResponse();
+            UltimatumTemplateValues templateValues = createTemplateValues();
+            String templateValuesAsJson = toJson(templateValues);
+            documentInfoResponse.setData(templateValuesAsJson);
+            documentInfoResponse.setAssetId(assetId);
+            documentInfoResponse.setTemplateName(templateName);
+            documentInfoResponse.setPath(renderedDocDir + renderedDocFileName);
+            return documentInfoResponse;
+        }
+
+        private UltimatumTemplateValues createTemplateValues() {
+            String companyName = prosecutionCase.getCompanyName();
+            String companyIncorporationNumber = prosecutionCase.getCompanyIncorporationNumber();
+
+            UltimatumTemplateValues templateValues = new UltimatumTemplateValues();
+            templateValues.setCompanyName(companyName);
+            templateValues.setCompanyNumber(companyIncorporationNumber);
+            templateValues.setTemplateRegistryAddress(templateRegistryAddress);
+            templateValues.setDefendantName("David Fraud");
+            return templateValues;
+        }
+
+        private String toJson(UltimatumTemplateValues templateValues)
+                        throws DocumentInfoCreationException {
+            ObjectMapper mapper = new ObjectMapper();
+            try {
+                return mapper.writeValueAsString(templateValues);
+            } catch (JsonProcessingException e) {
+                throw new DocumentInfoCreationException(
+                                "Could not serialise Document Info for Ultimatum");
```

Comment:
Can you add the prosecution case ID into this log so the document which failed to be created can be identified please.

Reply:
Fixed